### ISSUE TITLE
Auto BSL reset not working with USB adapters using Silabs CP2102N like Sonoff USB Plus Dongle

### ISSUE THREAD

**Issue by Hedda** [xyzroe/ZigStarGW-MT#2]
FYI, automatic BSL reset not working with USB adapters based on Silicon Labs CP2102N USB-to-UART chip and TI CC2652P Zigbee chip like the new SONOFF Zigbee 3.0 USB Dongle Plus by ITead

https://itead.cc/product/sonoff-zigbee-3-0-usb-dongle-plus/

Get error message:

`Timeout waiting for ACK/NACK after 'Synch (0x55 0x55)'`

Know issue in upstream cc2538-bsl by JelmerT but a workaround do exist by using boot reset code from Sonoff `uartlog.py` script:

https://github.com/JelmerT/cc2538-bsl/issues/113

![image](https://user-images.githubusercontent.com/6320001/142212064-d54263e8-dce5-4164-90cb-e2e0d66eeb5b.png)

**Comment by mercenaruss** [comment]
@Hedda need to use BSL button to enter bootloader,this is problem with hardware design.

**Comment by Hedda** [comment]
> @Hedda need to use BSL button to enter bootloader, this is problem with hardware design.

As I understand from the discussion in https://github.com/JelmerT/cc2538-bsl/issues/113 it is not really a problem with the hardware but either a bug in the CP2102N revision or more likely an issue with Microsoft Windows serial interface in combination with the bug in the CP2102N revision they use. It should however be possible to workaround those issues by adding additional delays, so this might get "fixed" in a future version of https://github.com/JelmerT/cc2538-bsl

**Comment by xyzroe** [comment]
When support will be implemented in JelmerT / cc2538-bsl then I will add.

**Comment by Hedda** [comment]
@xyzroe FYI,  JelmerT got it working via a new option attribute here -> https://github.com/JelmerT/cc2538-bsl/pull/114

**Comment by xyzroe** [comment]
> @xyzroe FYI, JelmerT got it working via a new option attribute here -> [JelmerT/cc2538-bsl#114](https://github.com/JelmerT/cc2538-bsl/pull/114)

OK. Thank you.
I'll do it at my leisure.

**Comment by xyzroe** [comment]
@Hedda Could you please check this video? https://youtu.be/-0JXHbYDFTs
Some guy flash sonoff stock using multitool with no changes or button pushes.

**Comment by Hedda** [comment]
> @Hedda Could you please check this video? https://youtu.be/-0JXHbYDFTs Some guy flash sonoff stock using multitool with no changes or button pushes.
> 
> From 3:30

Video was removed before I could see it.

**Comment by xyzroe** [comment]
Re-uploaded here https://m.youtube.com/watch?v=Y7XDC9oer5I

___________________
Sincerely, Semen!

**Comment by Hedda** [comment]
Yeah you can see in that video that he uses the "ITead_Sonoff_Zigbee-delay" branch from https://github.com/JelmerT/cc2538-bsl/pull/114

![image](https://user-images.githubusercontent.com/6320001/148793277-e3e48d52-5bfd-4586-8815-612a09c13600.png)

**Comment by xyzroe** [comment]
![Screen Shot 2022-01-10 at 6 20 53 PM](https://user-images.githubusercontent.com/6440415/148800651-0cb1c3e9-5b09-4cf7-ae76-80d3266013f8.png)
please watch again from 2:40

**Comment by RadioLed2000** [comment]
Just now I could flash sonoff stick with GW Multi tool without opening enclosure!!!
1. Plug it in the USB
2. Read NVRAM
3. Try to flash firmware, got an error
4. Unplug and plug again to USB port
5. Press Start to flash firmware and it successfelly flashed it

I don't know how it works, but it works)))

**Comment by TheNutHouse84** [comment]
@Hedda 
I was also able to flash my sonoff dongle by doing same as @RadioLed2000.

**Comment by fowl2** [comment]
Can confirm.

All I needed to do to flash without pressing the button was:
1. Plug in the dongle normally
2. click "write", receive "timeout" message
3. unplug+plug
4. click write again

**Comment by xbmcnut** [comment]
For me, I had to hold down the boot button for 20s before (and after) inserting the stick. 10s did not cut it multiple times.

**Comment by AlanADV** [comment]
Same here, only worked after holding the reset button for about 20 seconds after inserting the dongle.

**Comment by pitkaha** [comment]
> For me, I had to hold down the boot button for 20s before (and after) inserting the stick. 10s did not cut it multiple times. I kept getting `Timeout waiting for ACK/NACK after 'Synch (0x55 0x55)` prior.

Same. For the next one fighting with Sonoff ZBDongle-P lets make it simple. For me this worked:

1. Get ZigStar GW Multi and newest firmware [here](https://www.zigbee2mqtt.io/guide/adapters/#recommended).
2. Open the case, screws are PH0.
3. Open ZigStar.
4. Plug Sonoff into usb port while pushing down boot button and keep it pressed down for 20sec after power led has turned on.
5. Choose the right port and the firmware you downloaded on ZigStar.
6. Choose Erase, Write and Verify and click Write, the process bar should move quite quickly.
